### PR TITLE
[CR] [On hold] Rework mushrooms

### PR DESCRIPTION
**Pull request by Rail-Runner**
Rework mushrooms to make them to make it obvious which ones are edible or hallucinogenic, and some other changes. [CR] because I'm not sure if all of these changes should go in, in which case I'll revet them, but for me it still looks better than before.

List of changes:
1. Make mushrooms perishable and reduce their fun value. Raw mushrooms taste quite bad, actually, and they don't last forever.
2. Add dried hallucinogenic mushrooms, cooked mushrooms, sealed jar of mushroom soup and recipes for them.
3. Add mushrooms, dried mushrooms, cooked mushrooms and cooked morel mushrooms to some recipes.
4. Remove EATEN_HOT flags from mushrooms and add "RAW_WILDVEG" use action to mushroom.
5. Change recpe of dried mushrooms to be consistent with other dehydrating rcecipes.
6. Add recipe for mushroom soup.

### CODE REVIEW

**Comment by Zireael07** [comment]
+1 to making mushrooms perishable and for the dried/cooked recipes. And the soup!

File: data/json/items/comestibles.json [hunk 1]
```diff
@@ -6738,6 +6759,30 @@
...
+        "use_action" : "FUN_HALLU",
+        "stim" : 0,
+        "container" : "null",
+        "comestible_type" : "FOOD",
+        "symbol" : "%",
+        "quench" : -1,
+        "healthy" : 0,
+        "addiction_potential" : 0,
+        "nutrition" : 8,
+        "description" : "Some hallucinogenic mushrooms which have been dehydrated for storage.  Will still cause hallucinations if eaten.",
```
Comment (DeadLeaves) [inline data/json/items/comestibles.json]: Either make the descriptions be about the object in singular, or change the base name. 
``dried hallucinogenic mushrooms`` 
or
``A hallucinogenic mushroom which...``
I'd say go with option 2 as most items have their description in singular

**Comment by Coolthulhu** [comment]
What is the motivation for obvious shroom status?
If only the crafting menu trick making them easy to tell apart, then it could wait for #14950 which would make them meaningful.
It's a gimmicky mechanic, but if the status was consistently hidden from the player, it wouldn't necessarily be bad design to have poisonous shrooms.

**Comment by Darkyhard** [comment]
> it could wait for #14950 which would make them meaningful.

Are you currently working on it?

> What is the motivation for obvious shroom status?

Well, you could bypass this mechanic anyway by eating a single mushroom from a stack of them and then placing edible, poisonous and hallucinogenic mushrooms on different tiles of map. When you find some new mushrooms, you just check which mushrooms that you already have can stack with it, and place them in that stack.

Also, most people know which mushrooms are edible and which are poisonous, at least among most common of them. Even if they wouldn't, if they wouldn't have any source of information, they could only understand that only by eating them, no amount of simply harvesting them (like, survival skill) would help here. By the way, there are multiple species of edible, poisonous and hallucinogenic mushrooms, so even this wouldn't help much.

> it wouldn't necessarily be bad design to have poisonous shrooms.

They are poisonous for most characters and have no other uses, so they are usually just thrown away. The only use that I could think of for them is extracting their poison for use with some weapons or a bait, but the game currently doesn't allow this.

**Comment by Coolthulhu** [comment]
> Are you currently working on it?

Started, but then dropped it due to it depending on allergy rework (and this one in turn conflicting with my eating rewrite, already waiting for mergetest) to make sense. Nothing serious, just changing allergies to flags rather than materials.

> Well, you could bypass this mechanic anyway by eating a single mushroom from a stack of them and then placing edible, poisonous and hallucinogenic mushrooms on different tiles of map.

Now that shrooms will be perishable, they won't stack as nicely, meaning that identifying them by stacking would be harder.

> Also, most people know which mushrooms are edible and which are poisonous, at least among most common of them.

This could be represented by properties or something like that: rolling ID difficulty per spawned shroom or something like that.

> Even if they wouldn't, if they wouldn't have any source of information, they could only understand that only by eating them, no amount of simply harvesting them (like, survival skill) would help here.

I assume survival skill also means knowledge of things like poisonous food identification (using own skin, tasting etc.), just abstracted out. Shrooms need like 4-5 survival to ID and getting that much from harvesting alone would take a lot of time. While most people probably get it in an even less realistic way (butchery), butchery is one of those things that get nerfed quite often, so this could go away too.

> They are poisonous for most characters and have no other uses, so they are usually just thrown away.

But if they weren't that easy to ID, having to throw away ALL mushrooms just to get rid of poisonous ones would be a harder decision, at least for an early game character.

That said, I won't insist on keeping ID mechanics. While shroom ones are probably the best example of meaningful ID in DDA (they win by a large margin compared to book ID and fake medieval weapon ID), they still aren't perfect.

**Comment by mutt3x** [comment]
>Rework mushrooms to make them to make it obvious which ones are edible or hallucinogenic...

It would be more ideal if you read one of thouse survival skill books. By doing so you would gain the knowledge to spot what is what, or something in that way.

>Prevent spawning of poisonous mushroom. It's just useless now.
>They are poisonous for most characters and have no other uses..

Personally i dislike when things get removed from a game (someone did put it there), changing the mechanic is more accepted by my standards. Music CDs are useless too... but they add some kind of realism to household items (im sure you have CDs in your house), poisonous shrooms are a vital part of the forest loot/spawn/atmosphere... no?

>Also, most people know which mushrooms are edible and which are poisonous...

This, this can not be... we are masters of shroom gathering.
I have made a mushroom soup. I have gathered them myself. I hope me being a complete novice in mushroom gathering wont stop you from eating it.
HA, jokes aside, i would never pick a shroom from the ground if im not 100% sure its okey. Appearance is nothing but a trick in nature.
A lot of mushrooms look a like and you need a good eye, exsample is the Chlorophyllum molybdites, but what do i know.

**Comment by Zireael07** [comment]
There are many kinds of edible shrooms that are obvious - I mean it's obvious that they're good to eat.
At the same time, there's many poisonous shrooms which mimic other edible shrooms (which means most people IRL won't pick any of the "suspicious" shrooms since telling the edible from the poisonous is very difficult, you have to know what detail to look for for every such edible-poisonous mimic pair.

**Comment by Darkyhard** [comment]
> i would never pick a shroom from the ground if im not 100% sure its okey.

It's somewhat reflected by survival skill, actually. At 0 skill, you would pick up only most common mushrooms (so you get much less than you could get if your skill was higher), and as your skill increases, you gain knowledge about more rare mushroom species, so that you could harvest more of them.

**Comment by mutt3x** [comment]
>At 0 skill, you would pick up only most common and obviously edible mushrooms

so everybody starts with common knowledge about mushrooms... at that skill you only would know you picked up a **mushroom** and it looks edible, but thats not always necessary true. After all you are not an expert.
...
I just think they should be there and they should be unknown until you learn whats good... for the extra danger.
Maybe somebody would want to make something out of them in the future. Placing them as food traps for wildlife... something.

**Comment by Coolthulhu** [comment]
> Maybe somebody would want to make something out of them in the future.

I would disagree with this approach if I didn't just try to make something out of them. If there wereno prospects for poisonous mushrooms (and "maybe sometime" counts as no prospects), it would be better to phase them out rather than pretend they make sense.

That said, they may still make sense once flag passing is a thing.

**Comment by MisterFelixFox** [comment]
I disagree with the premise of this pull request -- 'most people' are not masters of identifying mushrooms. If I were to ask people around town to identify mushrooms, I imagine only a handful of them would be able to give a correct answer. There is even a feature in the code for you to be able to tell if a mushroom is poisonous once you reach Survival 2 (or 3? I'm not sure). It makes scavenging for food dangerous at low levels of Survival, and removing poisonous ones would just mean that eating random mushrooms isn't dangerous anymore (aside from the potential impacts and annoyances of hallucinations).

**Comment by Darkyhard** [comment]
> It makes scavenging for food dangerous at low levels of Survival

Nope. Usually it results either in ignoring all mushrooms at low skills, as there are other plants which are harvestable and not poisonous, or bypassing this skill check at all by using some tricks.

File: data/json/recipes/recipe_food.json [hunk 2]
```diff
@@ -3785,6 +3835,124 @@
...
+      [ "mushroom_cooked", 4 ],
+      [ "dry_mushroom", 4 ],
+      [ "morel_cooked", 4 ]
+    ]
+  ]
+},{
+  "type" : "recipe",
+  "result": "jar_soup_mushroom",
+  "category": "CC_FOOD",
+  "subcategory": "CSC_FOOD_MEAT",
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_food.json]: Meat? Should be veggy

**Comment by Darkyhard** [comment]
I think I'll leave it like this for now and then update later when flag passing makes it into the game.

**Comment by Rivet-the-Zombie** [comment]
@Darkyhard, everything other than the poisonous mushroom stuff is great; want to separate it out so we can use it now?

**Comment by Darkyhard** [comment]
> everything other than the poisonous mushroom stuff is great (and it seems that nobody's against it); want to separate it out so we can use it now?

The problem is that I will have to remove everything recipe-related, as it would make it easy to tell which mushrooms are edible, but adding mushrooms to recipes is the main reason for this PR.

**Pull request merged**